synapse/http/site.py

```diff
@@ -333,14 +333,28 @@ def _should_log_request(self) -> bool:
 
 
 class XForwardedForRequest(SynapseRequest):
-    def __init__(self, *args, **kw):
-        SynapseRequest.__init__(self, *args, **kw)
-
     """
     Add a layer on top of another request that only uses the value of an
     X-Forwarded-For header as the result of C{getClientIP}.
+
+    XXX: I think the right way to do this is with request.setHost().
     """
 
+    def __init__(self, *args, **kw):
+        SynapseRequest.__init__(self, *args, **kw)
+
+        forwarded_header = self.getHeader(b"x-forwarded-proto")
+        if forwarded_header is not None:
+            self._is_secure = forwarded_header.lower() == b"https"
```

Comment:
Any reason not to just set `isSecure` in these two places?

Reply:
sorry, which two places?

Reply:
Sorry for the opaque comment! I guess I'm asking why there's a separate `_is_secure` and then a method for returning that value, instead of setting the `isSecure` property directly. (As part of this I also wonder why we set this in `__init__` vs. `getClientIP` which is dynamic.)

Reply:
well, isSecure() is defined by the base class to be a callable method, not a property.

Reply:
Ah-ha! I didn't realize that this was part of the base-class! 👍